### PR TITLE
Fix log

### PR DESCRIPTION
**Pull request by maxnoe**
Uses server sent events to stream the log file, this was broken in the recent deployment and with this it should work again and be more stable.

I also improved authentication a bit, if you try to directly access a page via its url that needs authentication, you are redirected to the login and after login taken back to where you wanted to in the first place.

### CODE REVIEW

**Comment by maxnoe** [comment]
I'll deploy to test it in the afternoon

**Comment by maxnoe** [comment]
Now deployed.

File: shifthelper_webinterface/log.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                    'Log file was truncated, probably log rotation'
+                ), i)
+            else:
+                msg = f.read().decode()
+
+            yield build_sse(msg, i)
+            sleep(1)
+
+
+def build_sse(message, id_=None):
```
Comment (dneise) [inline shifthelper_webinterface/log.py:44]: I do not know what `sse` stands for ... (and I was also unable to find it by quickly googling ... https://en.wikipedia.org/wiki/SSE)

maybe this name could be refactored ... unless it is a fixed term one simply has to learn.

Comment (maxnoe) [reply shifthelper_webinterface/log.py:44]: The full term is explained in the docstring and its the second in the list of the wikipedia article you linked.

> Server-sent events, a technology to push content to web clients

**Pull request merged**